version: '3'
services:
    odoo:
        build: .
        container_name: odoo
        restart: always
        volumes:
            - ./addons:/mnt/extra-addons
            - ./config/odoo:/etc/odoo
            - ./odoo-web-data:/var/lib/odoo
        ports:
            - "8069:8069"
            - "8888:8888"
        command:  bash -c "/usr/bin/python3 -m debugpy --listen 0.0.0.0:8888 /usr/bin/odoo --db_user=odoo --db_host=db --db_password=odoo"
        depends_on:
            - "db"
        networks:
          - odoo_network

    nginx:
        container_name: nginx
        image: nginx:latest
        restart: unless-stopped
        ports:
            - 80:80
            - 443:443
        volumes:
            - ./config/nginx/conf:/etc/nginx/conf.d/:rw
            - ./certbot/conf:/etc/letsencrypt
            - ./certbot/www:/var/www/certbot
        depends_on:
            - "odoo"
        networks:
            - odoo_network

    certbot:
        image: certbot/certbot
        volumes:
            - ./certbot/conf:/etc/letsencrypt
            - ./certbot/www:/var/www/certbot
        command: certonly --webroot -w /var/www/certbot --force-renewal --email danielpontesnery@gmail.com -d erpmontec.empresarial.ws --agree-tos
        depends_on:
            - nginx
        networks:
          - odoo_network

    db:
      image: postgres:13
      restart: always
      container_name: postgresql
      environment:
        - POSTGRES_PASSWORD=odoo
        - POSTGRES_USER=odoo
        - POSTGRES_DB=postgres
        - PGDATA=/var/lib/postgresql/data/pgdata
      ports:
        - "127.0.0.1:9432:5432"
      volumes:
        - ./odoo-db-data:/var/lib/postgresql/data/pgdata
      logging:
        driver: json-file
        options:
          max-size: "30m"
          max-file: "10"

volumes:
  odoo-web-data:
  odoo-db-data:
